yt/frontends/enzo_e/fields.py

```diff
@@ -143,17 +143,34 @@ def setup_particle_mass_field(self, ptype):
                 constants = (constants,)
             names = [c[0] for c in constants]
 
+        group_list = nested_dict_get(
+            self.ds.parameters, ("Particle", ptype, "group_list"), default=()
+        )
+
         if "mass" in names:
```

Comment:
As far as I can tell, `names` just contains the names of the particle constants. So what happens when a particle type does not have a "mass" or "density" constant, but rather, has an "attribute array" for these quantities?

Reply:
Is this a possibility? If so, could you provide me with a sample dataset that does this?